sound/soc/sof/intel/hda-dsp.c

```diff
@@ -779,17 +779,27 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	const struct sof_intel_dsp_desc *chip = hda->desc;
 	struct hdac_bus *bus = sof_to_bus(sdev);
+	bool imr_lost = false;
 	int ret, j;
 
 	/*
-	 * The memory used for IMR boot loses its content in deeper than S3 state
-	 * We must not try IMR boot on next power up (as it will fail).
-	 *
+	 * The memory used for IMR boot loses its content in deeper than S3
+	 * state on CAVS platforms.
```

Comment:
@ujfalusi did you test S3 suspend on cavs to confirm this? I feel like maybe its not necessary to do this distinction for cavs and ace at all and just always do a cold boot from S3 for all platforms

Reply:
@ranj063 it's hard to test/prove something like this, I think the 'less-risky' course of action is to keep cavs as is and only use cold boot from S3 for MTL+

Reply:
@ranj063, yes, I have tested S3 suspend on TGL (on upx-i11)

Reply:
and the reason why we ended up with the `sdev->system_suspend_target > SOF_SUSPEND_S3` was that users reported issues with only suspend to disk, [deep] state has been working fine.